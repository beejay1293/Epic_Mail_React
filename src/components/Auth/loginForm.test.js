import React from 'react';
import sinon from 'sinon'
import store from '../../store';
import { mount, shallow, render } from 'enzyme';
import LoginForm, { Login } from './loginForm';
describe('Login form', () => {
  it('should render correctly the login form', () => {
    const Form = shallow(<LoginForm store={store}/>);
  
    expect(Form).toMatchSnapshot();
  });
});

describe('test login methods', () => {
  sinon.spy(Login.prototype, 'handleInput');
  sinon.spy(Login.prototype, 'handleSubmit');

  const props = {
    isAuthenticated: true,
    isLoading: true,
    user: {},
    errors: {},
    auth: () => {}
  }

  const enzymeWrapper = shallow(<Login {...props} />);

 describe('Login', () => {
   let wrapper;
   beforeEach(() => {
     wrapper = enzymeWrapper;
   })

   it('test the submit button', () => {
     const form = wrapper.find('.login-form')
     const mockedEvent = { target: {}, preventDefault: () => {} }
     form.simulate('submit', mockedEvent)
     expect(Login.prototype.handleSubmit.called).toBe(true);
   })

   it('should test the Input field', () => {
      const input = wrapper.find('input').first();
      const e = { target: {}, preventDefault: () => {}};

      input.simulate('change', e)
      expect(Login.prototype.handleInput.called).toBe(true);
   })

    it('should test the Input field', () => {
      const input = wrapper.find('input').at(1);
      const e = { target: {}, preventDefault: () => {}};

      input.simulate('change', e)
      expect(Login.prototype.handleInput.called).toBe(true);
   })
 })


})

describe('test login methods', () => {
  const props = {
    isAuthenticated: true,
    isLoading: false,
    user: {},
    errors: {},
    auth: () => {}
  }
  const enzymeWrapper = shallow(<Login {...props} />);

  it('should test wrapper', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  })

})