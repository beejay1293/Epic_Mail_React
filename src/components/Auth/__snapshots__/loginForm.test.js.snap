// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form should render correctly the login form 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Login)
    auth={[Function]}
    errors={Object {}}
    isAuthenticated={false}
    isLoading={false}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    user={Object {}}
  />
</ContextProvider>
`;
